zimui/src/stores/main.ts

```diff
@@ -1,52 +1,94 @@
   state: () =>
     ({
       channelData: null,
       isLoading: false,
       errorMessage: '',
+      error: null,
     }) as RootState,
   getters: {},
   actions: {
     async fetchChannel() {
       this.isLoading = true
       this.errorMessage = ''
-      return axios.get('./channel.json').then(
-        (response) => {
-          this.isLoading = false
-          this.channelData = response.data as Channel
-        },
-        (_) => {
+      try {
+        const response = await axios.get('./channel.json')
+        this.isLoading = false
+        this.channelData = response.data as Channel
+      } catch (error) {
+        if (error instanceof AxiosError) {
           this.isLoading = false
           this.channelData = null
-          this.errorMessage = 'Failed to load channel data'
-        },
-      )
+          this.handleAxiosError(error);
+          if (!axios.isAxiosError(error)) {
+            this.errorMessage = 'Failed to load channel data';
+          }
+          this.error = error // Set axios error to store
+        }
+      }
     },
     async fetchTopic(slug: string) {
       this.isLoading = true
       this.errorMessage = ''
-      return axios.get('./topics/' + slug + '.json').then(
-        (response) => {
-          this.isLoading = false
-          return response.data as Topic
-        },
-        (_) => {
+      try {
+        const response = await axios.get('./topics/' + slug + '.json')
+        this.isLoading = false
+        return response.data as Topic
+      } catch (error: unknown) {
+        if (error instanceof AxiosError) {
           this.isLoading = false
           this.channelData = null
-          this.errorMessage = 'Failed to load node ' + slug + ' data'
+          this.handleAxiosError(error);
+          if (!axios.isAxiosError(error)) {
+            this.errorMessage = 'Failed to load node ' + slug + ' data';
+          }
+          this.error = error // Set axios error to store
           return null
-        },
-      )
+        }
+      }
+    },
+    handleAxiosError(error: AxiosError<object>) {
+      this.isLoading = false;
+      this.channelData = null;
+      if (axios.isAxiosError(error)) {
+        if (error.response) {
+          const status = error.response.status;
+          switch (status) {
+            case 400:
+              this.errorMessage = 'Bad Request: The server could not understand the request due to invalid syntax.';
+              break;
+            case 401:
+              this.errorMessage = 'Unauthorized: Authentication is required and has failed or has not yet been provided.';
+              break;
+            case 403:
+              this.errorMessage = 'Forbidden: The server understood the request but refuses to authorize it.';
+              break;
+            case 404:
+              this.errorMessage = 'Not Found: The requested resource could not be found on the server.';
+              break;
+            case 500:
```

Comment:
This adds no value to the generic message "An error occured"

Reply:
you mean `case 500` or `case 404`
Because both of them seem important to me,
`case 404` when the requested data is not present.
`case 500` this might arise in the event of network error, compiler error, docker error etc.

Reply:
I would say we never know what the user interprets from the error message, I guess we can keep the statement until some user enquires OR do you recommend removing the line? Because, for developers, it might narrow down the potential reasons for any error they might be dealing with. So, what do you say?

Reply:
OK, let's keep this.